go.mod

```diff
@@ -14,74 +14,73 @@ require (
-	k8s.io/gengo v0.0.0-20200413195148-3a45101e95ac
+	k8s.io/cluster-bootstrap v0.19.6
+	k8s.io/code-generator v0.19.6
+	k8s.io/component-base v0.19.6
+	k8s.io/gengo v0.0.0-20200428234225-8167cfdcfc14
 	k8s.io/helm v2.16.1+incompatible
 	k8s.io/klog v1.0.0
-	k8s.io/kube-aggregator v0.18.10
-	k8s.io/kube-openapi v0.0.0-20200410145947-bcb3869e6f29 // release-1.18
-	k8s.io/kube-scheduler v0.18.10
-	k8s.io/kubelet v0.18.10
-	k8s.io/metrics v0.18.10
-	k8s.io/utils v0.0.0-20200619165400-6e3d28b6ed19
-	sigs.k8s.io/controller-runtime v0.6.3
-	sigs.k8s.io/controller-tools v0.3.0
+	k8s.io/kube-aggregator v0.19.6
+	k8s.io/kube-openapi v0.0.0-20201221124747-75e88872edcf // keep this value in sync with k8s.io/apiserver
+	k8s.io/kube-scheduler v0.19.6
+	k8s.io/kubelet v0.19.6
+	k8s.io/metrics v0.19.6
+	k8s.io/utils v0.0.0-20200912215256-4140de9c8800
+	sigs.k8s.io/controller-runtime v0.7.0
+	sigs.k8s.io/controller-tools v0.4.1
 	sigs.k8s.io/yaml v1.2.0
 )
 
 replace (
 	github.com/emicklei/go-restful => github.com/emicklei/go-restful v2.9.5+incompatible // keep this value in sync with k8s.io/apiserver
 	github.com/envoyproxy/go-control-plane => github.com/envoyproxy/go-control-plane v0.9.4
-	github.com/prometheus/client_golang => github.com/prometheus/client_golang v0.9.2
-	google.golang.org/grpc => google.golang.org/grpc v1.26.0 // keep this value in sync with k8s.io/apiserver
-	k8s.io/api => k8s.io/api v0.18.10
-	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.18.10
-	k8s.io/apimachinery => k8s.io/apimachinery v0.18.10
-	k8s.io/apiserver => k8s.io/apiserver v0.18.10
-	k8s.io/client-go => k8s.io/client-go v0.18.10
-	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.18.10
-	k8s.io/code-generator => k8s.io/code-generator v0.18.10
-	k8s.io/component-base => k8s.io/component-base v0.18.10
+	github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.4.1
+	github.com/prometheus/client_golang => github.com/prometheus/client_golang v1.7.1 // keep this value in sync with k8s.io/client-go
+	google.golang.org/grpc => google.golang.org/grpc v1.27.0 // keep this value in sync with k8s.io/apiserver
+	k8s.io/api => k8s.io/api v0.19.6
+	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.19.6
+	k8s.io/apimachinery => k8s.io/apimachinery v0.19.6
+	k8s.io/apiserver => k8s.io/apiserver v0.19.6
+	k8s.io/client-go => k8s.io/client-go v0.19.6
+	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.19.6
+	k8s.io/code-generator => k8s.io/code-generator v0.19.6
+	k8s.io/component-base => k8s.io/component-base v0.19.6
 	k8s.io/helm => k8s.io/helm v2.13.1+incompatible
-	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.18.10
-	k8s.io/kube-openapi => github.com/gardener/kube-openapi v0.0.0-20200831104310-b5db060350c7
-	sigs.k8s.io/controller-runtime => github.com/gardener/controller-runtime v0.6.3-gardener.1
-	sigs.k8s.io/structured-merge-diff/v3 => sigs.k8s.io/structured-merge-diff/v3 v3.0.1-0.20201124164700-f5fd4ea1e4c9
```

Comment:
I see that the `sigs.k8s.io/structured-merge-diff` replace is now removed but as far I see the upstream fix that we need https://github.com/kubernetes-sigs/structured-merge-diff/pull/173 is not included in the current version of `sigs.k8s.io/structured-merge-diff`.

```
$ go list -mod=mod -m all | grep sigs.k8s.io/structured-merge-diff | awk '{print $2}'
v4.0.1
```

As far as I see v4.0.1 does not contain https://github.com/kubernetes-sigs/structured-merge-diff/pull/173 and this can regress https://github.com/gardener/gardener/issues/2977. I can check whether I can file a cherry-pick PR about it.

Reply:
Yes, good point. I wasn't sure about this. It's still on my list for testing.

Reply:
Ok, just confirmed. This currently regresses #2977, so we probably need to update our fork again until we have an upstream cherry-pick.

Reply:
Ref https://github.com/kubernetes-sigs/structured-merge-diff/pull/182